File: android/QrCode/app/src/main/java/com/example/qrcode/MainActivity.java
package com.example.qrcode;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.zxing_button).setOnClickListener(this);
        findViewById(R.id.zxing_android_button).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.zxing_button) {
            Snackbar.make(v, "ZXing", Snackbar.LENGTH_SHORT).show();
        }
        else if (v.getId() == R.id.zxing_android_button) {
            Snackbar.make(v, "ZXing Android", Snackbar.LENGTH_SHORT).show();
        }
    }
}

File: android/EspressoTest/app/src/main/java/com/example/espressotest/MainActivity.kt
package com.example.espressotest

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.example.espressotest.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main )

        binding.button.setOnClickListener {
            binding.text.text = "Button Clicked"
        }
    }
}